### PR TITLE
Add destroy server group ops and other minor bug fixes

### PR DESCRIPTION
**Pull request by milismsft**
Add destroy server group ops. This payload was reviewed previously; the only addition is the test and removing some commented out lines in the validation class.
Add type for the deployment. Updating the python tests to reflect these changes is a separate task.
Cleanup resources that are created as part of the create server group in case of a failure during the creation.
Add retries support when deleting a resource in Azure (work around for random timeouts).
Fix an assertion in clouddriver console when the application has no clusters.

@JargoonPard @scotmoor PTAL; I'm looking to submit A PR with these changes later tonight.

### CODE REVIEW

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy [hunk 1]
```diff
@@ -154,6 +155,28 @@ public class AzureComputeClient extends AzureBaseClient {
...
+    // The API call might return a timout exception or some other Azure CloudException that is not
+    //   related to the operation we are trying to execute; retry couple of times and if the final
+    //   retry fails then rethrow
+    long operationRetry = 0
+    while (operationRetry < AZURE_ATOMICOPERATION_RETRY) {
+      try {
+        operationRetry ++
+        result = this.client.getVirtualMachineScaleSetsOperations()?.delete(resourceGroupName, serverGroupName)
+        operationRetry = AZURE_ATOMICOPERATION_RETRY
+      } catch (Exception e) {
```
Comment (scotmoor) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy]: What if the exception is a cloud exception and the status code is a 404? Would you want to keep trying to delete it if it's already gone?

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy [hunk 2]
```diff
@@ -184,13 +184,61 @@ class AzureNetworkClient extends AzureBaseClient {
...
+    // The API call might return a timout exception or some other Azure CloudException that is not
+    //   related to the operation we are trying to execute; retry couple of times and if the final
+    //   retry fails then rethrow
+    long operationRetry = 0
+    while (operationRetry < AZURE_ATOMICOPERATION_RETRY) {
+      try {
+        operationRetry ++
+        result = client.getLoadBalancersOperations().delete(resourceGroupName, loadBalancerName)
+        operationRetry = AZURE_ATOMICOPERATION_RETRY
+      } catch (Exception e) {
```
Comment (scotmoor) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy]: Same comment about handling 404s

**Comment by ghost** [comment]
LGTM

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy (hunk 1 above)
Comment (milismsft) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureComputeClient.groovy]: I checked the Azure JSDK implementation and for post or delete operations they don't handle 404:
com.microsoft.azure/azure-client-runtime/1.0.0-beta1/79374d64473fc93e2a2478e7b374dd5d289c98b1/azure-client-runtime-1.0.0-beta1-sources.jar!/com/microsoft/azure/AzureClient.java:270
        if (statusCode != 200 && statusCode != 202 && statusCode != 204) {
            CloudException exception = new CloudException(statusCode + " is not a valid polling status code");
            exception.setResponse(response);

**Pull request merged**